ethereum/contracts/EtherVault.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// Holds ETH on behalf of ETHApp
+contract EtherVault is Ownable {
+    event Deposit(address account, address origin, uint256 amount);
+    event Withdraw(address account, address recipient, uint256 amount);
+
+    receive() external payable {
+        revert("Must use deposit function");
+    }
+
+    // Accepts ETH from the caller.
+    function deposit() 
+        external
+        payable
+        onlyOwner
+    {
+        emit Deposit(msg.sender, tx.origin, msg.value);
```

Comment:
Use of `tx.origin` isn't recommended: https://consensys.github.io/smart-contract-best-practices/development-recommendations/solidity-specific/tx-origin/

Rather pass the depositee address as a parameter to `deposit`.

Then the EthApp can call `vault.deposit(msg.sender)`, because in that context `msg.sender` will be the account which interacted with EthApp